src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -117,5 +147,2211 @@ public void TexlDateTableFunctions_Negative(string expression)
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "RoundUp(1234.567, T)",
+                TestUtils.DT("*[Result:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "RoundUp(X, T)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "RoundUp(X, 4)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsRoundDown()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[digits:n]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                "RoundDown(1234.567, 4)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "RoundDown(1234.567, T)",
+                TestUtils.DT("*[Result:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "RoundDown(X, T)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "RoundDown(X, 4)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+        }
+
+        [Theory]
+        [InlineData("Sort(Table, A)", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table, A, \"Ascending\")", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table, A, \"Descending\")", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table, B)", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table, C)", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table!A, A)", "*[A:n]")]
+        [InlineData("Sort(Table, B & \"hello\")", "*[A:n, B:s, C:b, D:*[X:n]]")]
+        [InlineData("Sort(Table!B, Left(B, 2))", "*[B:s]")]
+        [InlineData("Sort(Table2, D!X & \"hello\")", "*[A:n, B:s, C:b, D:![X:n]]")]
+        [InlineData("Sort(Table2, D!X + 2)", "*[A:n, B:s, C:b, D:![X:n]]")]
+        public void TexlFunctionTypeSemanticsSort(string script, string expectedType)
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("Table", new TableType(TestUtils.DT("*[A:n, B:s, C:b, D:*[X:n]]")));
+            symbol.AddVariable("Table2", new TableType(TestUtils.DT("*[A:n, B:s, C:b, D:![X:n]]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                script,
+                TestUtils.DT(expectedType),
+                symbol);
+        }
+
+        [Theory]
+        [InlineData("Sqrt(1234.567)", "n")]
+        [InlineData("Sqrt(T)", "*[A:n]")]
+        [InlineData("Sqrt(T2!Value)", "*[Value:n]")]
+        public void TexlFunctionTypeSemanticsSqrt(string script, string expectedType)
```

Comment:
>TexlFunctionTypeSemanticsSqrt [](http://example.com/codeflow?start=20&length=29)

There are some other functions (like Sqrt) for which we should have tests for the ConsistentOneColumnTableResult flag. Basically every function on libraries\Microsoft.PowerFx.Core\Texl\BuiltinFunctionsCore.cs with a "T" suffix should have tests for that.

Reply:
This will be tracked via https://github.com/microsoft/Power-Fx/issues/710